spectator-web-spring/src/main/java/com/netflix/spectator/controllers/MetricsController.java

```diff
@@ -0,0 +1,229 @@
+ * limitations under the License.
+ */
+
+package com.netflix.spectator.controllers;
+
+import com.netflix.spectator.api.Counter;
+import com.netflix.spectator.api.DistributionSummary;
+import com.netflix.spectator.api.Gauge;
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.LongTaskTimer;
+import com.netflix.spectator.api.Measurement;
+import com.netflix.spectator.api.Meter;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spectator.api.Timer;
+
+import com.netflix.spectator.controllers.filter.ChainedMeasurementFilter;
+import com.netflix.spectator.controllers.filter.MeasurementFilter;
+import com.netflix.spectator.controllers.filter.PrototypeMeasurementFilter;
+import com.netflix.spectator.controllers.filter.TagMeasurementFilter;
+import com.netflix.spectator.controllers.model.ApplicationRegistry;
+import com.netflix.spectator.controllers.model.MetricValues;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+
+
+/**
+ * Provides an HTTP endpoint for polling spectator metrics.
+ */
+@RequestMapping("/spectator/metrics")
+@RestController
+@ConditionalOnExpression("${spectator.endpoint.enabled:false}")
+public class MetricsController {
+  @Autowired
+  private Registry registry;
+
+  @Value("${spectator.applicationName:}")
+  private String applicationName;
+
+  @Value("${spectator.applicationVersion:}")
+  private String applicationVersion;
+
+  @Value("${spectator.endpoint.tagFilter.meterNameRegex:.*}")
+  private String meterNameRegex;
+
+  @Value("${spectator.endpoint.tagFilter.tagNameRegex:}")
+  private String tagNameRegex;
+
+  @Value("${spectator.endpoint.tagFilter.tagValueRegex:}")
+  private String tagValueRegex;
+
+  @Value("${spectator.endpoint.prototypeFilter.path:}")
+  private String prototypeFilterPath;
+
+  private MeasurementFilter defaultMeasurementFilter = null;
+
+  /**
+   * The default measurement filter is configured through properties.
+   */
+  public MeasurementFilter getDefaultMeasurementFilter() throws IOException {
+    if (defaultMeasurementFilter != null) {
+      return defaultMeasurementFilter;
+    }
+
+    if (!prototypeFilterPath.isEmpty()) {
+      defaultMeasurementFilter = PrototypeMeasurementFilter.loadFromPath(
+          prototypeFilterPath);
+    } else {
+      defaultMeasurementFilter = new TagMeasurementFilter(
+          meterNameRegex, tagNameRegex, tagValueRegex);
+    }
```

Comment:
Just as an FYI, for our internal plugins that do something similar we have a single property that sets a filter expression. The expression is the query portion of the Atlas stack language: https://github.com/Netflix/atlas/wiki/Stack-Language#query. 

The main reason is that regex is hard to precisely target with dimensional ids. Also we have found that quite often filters are used to exclude so it is easier for a user to express what should not come through rather than or in addition to what should. I'm not saying use an Atlas expression and it probably isn't familiar to the users this is targeting , but I think it is worth considering how cumbersome the three regexes will be for expressing common filters.